sdk/core/core-client-lro-rest/src/getLongRunningHelper.ts

```diff
@@ -0,0 +1,183 @@
+ * Typically this is described in the Swagger using the extension `x-ms-long-running-operation-options`.
+ * @returns - A poller object to poll for operation state updates and eventually get the final response.
+ */
+export function getLongRunningPoller<TResult extends HttpResponse>(
+  client: Client,
+  initialResponse: TResult,
+  pollerOptions: LROPollerOptions,
+  finalStateVia: FinalStateVia
+): PollerLike<PollOperationState<TResult>, TResult>;
+export function getLongRunningPoller<TResult extends HttpResponse>(
+  client: Client,
+  initialResponse: TResult,
+  pollerOptionsOrFinalState?: LROPollerOptions | FinalStateVia,
+  finalStateViaOrNothing?: FinalStateVia
+): PollerLike<PollOperationState<TResult>, TResult> {
+  // Default values
+  let pollerOptions: LROPollerOptions = {};
+  let finalStateVia: FinalStateVia | undefined;
+
+  // Figure out which override we are handling
+  if (isFinalStateVia(pollerOptionsOrFinalState)) {
+    finalStateVia = pollerOptionsOrFinalState;
+  } else if (finalStateViaOrNothing) {
+    finalStateVia = finalStateViaOrNothing;
+  }
+
+  if (isLROPollerOptions(pollerOptionsOrFinalState)) {
+    pollerOptions = pollerOptionsOrFinalState;
+  }
+
+  // Translating the rest response into a response that the LRO engine understands
+  const initialLROResponse = getLroResponse(initialResponse);
+
+  const poller: LRO<TResult> = {
+    requestMethod: initialResponse.request.method,
+    requestPath: initialResponse.request.url,
+    sendInitialRequest: async (initializeState) => {
+      // In the case of Rest Clients we are building the LRO poller object from a response that's the reason
+      // we are not triggering the initial request here, just extracting the information from the
+      // response we were provided.
+      initializeState(initialLROResponse.rawResponse, initialLROResponse.flatResponse);
+      return initialLROResponse;
+    },
+    sendPollRequest: async (config, path) => {
+      // This is the callback that is going to be called to poll the service
+      // to get the latest status. We use the client provided and the polling path
+      // which is an opaque URL provided by caller, the service sends this in one of the following headers: operation-location, azure-asyncoperation or location
+      // depending on the lro pattern that the service implements. If non is provided we default to the initial path.
+      const response = (await client
+        .pathUnchecked(path ?? initialResponse.request.url)
+        .get()) as TResult;
+
+      // Translating the rest response into a response that the LRO engine understands
+      const lroResponse = getLroResponse(response);
+
+      // Hand off the information we have to the LRO engine to continue polling
+      return createGetLROState<TResult>(
+        poller,
+        config,
+        finalStateVia
+      )(lroResponse.rawResponse, lroResponse.flatResponse);
+    },
+    retrieveAzureAsyncResource: async (path) => {
```

Comment:
Similar feedback here - is it possible that we can construct things such that `retrieveAzureAsyncResource` is never called with a falsy path?

Reply:
`retrieveAzureAsyncResource` has been removed